File: everyday/src/main/java/com/landleaf/everyday/dao/AppDatabase.java
package com.landleaf.everyday.dao;

import androidx.room.Database;
import androidx.room.RoomDatabase;

import com.landleaf.everyday.bean.User;

/**
 * Author：Lei on 2020/12/10
 */
@Database(entities = {User.class}, version = 1)
public abstract class AppDatabase extends RoomDatabase {
    public abstract UserDao userDao();
}

File: kotlin/src/main/java/com/landleaf/kotlin/AppDatabase.kt
package com.landleaf.kotlin

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase

/**
 * Author：Lei on 2021/5/16
 */
@Database(version = 1, entities = [User::class])
abstract class AppDatabase : RoomDatabase() {

    companion object {
        private var instance: AppDatabase? = null
        @Synchronized
        fun getDataBase(context: Context): AppDatabase {
            instance?.let {
                return it
            }
            return Room.databaseBuilder(context.applicationContext, AppDatabase::class.java, "db_name")
                    .build()
                    .apply { instance = this }
        }
    }

    abstract fun userDao(): UserDao
}